app/controllers/api/v1/mappings_controller.rb

```diff
@@ -84,16 +84,18 @@ def authorize_with_policy
   # @return [ActiveRecord::Relation]
   ###
   def filter
-    # Always show only the mappings for the current user's organization
-    mappings = current_user.organization.mappings
-
-    # Filter by current user
-    mappings = mappings.where(user: current_user) if params[:filter] != "all"
-
-    # Return an ordered list
+    mappings = initial_mappings
+    mappings = mappings.where(user: current_user) if params[:filter] && params[:filter] != "all"
     mappings.order(:title)
   end
 
+  ###
+  # @description: Instantiate the list of mappings depending on the user roles
+  ###
+  def initial_mappings
+    current_user.super_admin? || current_user.profile_admin? ? Mapping.all : current_user.organization.mappings
```

Comment:
`User#organization` is an optional association. The `current_user.organization.mappings` expression will throw a `NoMethodError`, if `organization` is `nil`.

Reply:
@excelsior I added a fix but, just to clarify, it's optional because if a user has "super admin" role, it does not need to belong to an organization. That's the only case when a user can be without an organization.

Reply:
I re-seeded the DB before testing, and some of the users created from the seeds didn't belong to any organization. None of those were super admins, though. The requirement on organization presence doesn't seem to be enforced. Let's adjust the seeds at least?

Also, the `initial_mappings` method returns either a relation or an array, which isn't good, because this piece of code would fail for an array:

```ruby
def filter
  mappings = initial_mappings
  mappings = mappings.where(user: current_user) if params[:filter] && params[:filter] != "all"
  mappings.order(:title) # The `Array#order` method doesn't exist
end
```

I'd suggest replacing `[]` with `Mapping.none` in `initial_mappings`.

Reply:
@excelsior done